### PR TITLE
Update effective area and resolution curves

### PR DESCRIPTION
**Pull request by mikael10j**
Effective area:
- Compute effective area with simulation information
- Compute effective area for:
  - triggered events (maximum effective area)
  - reconstructed events (after selection cuts)
  - after discrimination (in case of full analysis)

Resolution curves: 
- Compute resolution curves for:
  - reconstructed events (after selection cuts)
  - after discrimination (in case of full analysis)

### CODE REVIEW

File: gammaboard.py [hunk 1]
```diff
@@ -40,29 +40,71 @@ def load_number_run(experiment, experiments_directory):
...
+    return trig_events
 
-    return num_run * number_event_per_run
+
+def load_run_config(experiment, experiments_directory):
+    assert experiment in os.listdir(experiments_directory)
+    file = experiments_directory + '/' + experiment + '/' + experiment + '.h5'
+    num_events = 0
+    spectral_index = []
+    max_scatter_range = -np.inf
```
Comment (vuillaut) [inline gammaboard.py]: Je ne comprends pas ces initialisations et les vérifications qui s'en suivent.

File: gammaboard.py [hunk 2]
```diff
@@ -40,29 +40,71 @@ def load_number_run(experiment, experiments_directory):
...
+            num_events += row['num_showers'] * row['shower_reuse']
+            spectral_index.extend([row['spectral_index']])
+            max_scatter_range = row['max_scatter_range'] if max_scatter_range < row['max_scatter_range'] else max_scatter_range
+            energy_range_max = row['energy_range_max'] if energy_range_max < row['energy_range_max'] else energy_range_max
+            energy_range_min = row['energy_range_min'] if energy_range_min > row['energy_range_min'] else energy_range_min
+            min_alt = row['min_alt'] if min_alt > row['min_alt'] else min_alt
+            max_alt = row['max_alt'] if max_alt < row['max_alt'] else max_alt
+        assert np.alltrue(np.array(spectral_index) == spectral_index[0]), \
+            'Cannot deal with different spectral index for the experiment ({})'.format(experiment)
+        print(min_alt)
```
Comment (vuillaut) [inline gammaboard.py]: prints to be removed

**Review by vuillaut — CHANGES_REQUESTED** [review]
Super, merci pour ces changements.
Deux petits points à discuter.

File: gammaboard.py (hunk 1 above)
Comment (mikael10j) [reply gammaboard.py]: résolu par 0a9624a

File: gammaboard.py (hunk 2 above)
Comment (mikael10j) [reply gammaboard.py]: résolu par 0a9624a

File: gammaboard.py [hunk 3]
```diff
@@ -66,11 +53,11 @@ def load_run_config(experiment, experiments_directory):
...
-    max_scatter_range = -np.inf
-    energy_range_max = -np.inf
-    energy_range_min = np.inf
-    min_alt = np.inf
-    max_alt = -np.inf
+    max_scatter_range = []
+    energy_range_max = []
+    energy_range_min = []
+    min_alt = []
+    max_alt = []
```
Comment (vuillaut) [inline gammaboard.py:62]: No need to store all values, I would have proposed something along the following.
But I guess there are not many values anyway to store so we won't run into an issue here.


```
    spectral_index = None
    max_scatter_range = None
    
    
    for row in run_config:
        spectral_index = row['spectral_index'][0] if spectral_index is None else spectral_index
        max_scatter_range = row['max_scatter_range'][0] if max_scatter_range is None else max_scatter_range
    assert spectral_index == row['spectral_index'][0]
    assert max_scatter_range == row['max_scatter_range'][0]
```

Comment (vuillaut) [reply gammaboard.py:62]: Or even 
```
spectral_index = run_config[0]['spectral_index'][0]
...
for row in run_config[1:]:
    assert spectral_index = row['spectral_index'][0]
    ...
```

**Comment by vuillaut** [comment]
@mikael10j you tested on lappuds ?

**Comment by mikael10j** [comment]
Nope, only on my computer.

File: gammaboard.py [hunk 4]
```diff
@@ -30,39 +31,74 @@ def load_data(experiment, experiments_directory):
...
 
-def load_number_run(experiment, experiments_directory):
+def load_trig_events(experiment, experiments_directory):
     assert experiment in os.listdir(experiments_directory)
 
     try:
-        num_run = int(pd.read_hdf(experiments_directory + '/' + experiment + '/' + experiment + '.h5',
-                           key='runs',
-                           )['num'][0])
+        trig_events = pd.read_hdf(experiments_directory + '/' + experiment + '/' + experiment + '.h5',
```
Comment (vuillaut) [inline gammaboard.py:43]: These events could actually be a sub-selection of triggered events.
Right now, for example, the events not passing the reconstruction process are not save in the result file.
But I agree on the principle, in a perfect world (a.k.a. the future), all triggered events would end-up in the DL2 file with a tag saying reconstructed or not.

Comment (mikael10j) [reply gammaboard.py:43]: I don't understand your point. For now `trig_events` contains the list of all triggered event's energy, to compute the maximum reachable effective area.

**Review by vuillaut — APPROVED** [review]

File: gammaboard.py (hunk 4 above)
Comment (vuillaut) [reply gammaboard.py:43]: With the files created from GammaLearn yes, but this is not necessarily the case.
My example was referring to lstchain. 
In any case, this is not a big deal, just a naming convention.

**Pull request merged**